server/service/hosts.go

```diff
@@ -2160,3 +2143,53 @@ func (svc *Service) HostLiteByID(ctx context.Context, id uint) (*fleet.HostLite,
 
 	return host, nil
 }
+
+func hostListOptionsFromFilters(filter *map[string]interface{}) (*fleet.HostListOptions, *uint, error) {
+	var labelID *uint
+
+	if filter == nil {
+		return nil, nil, nil
+	}
+
+	opt := fleet.HostListOptions{}
+
+	for k, v := range *filter {
+		switch k {
+		case "label_id":
+			if l, ok := v.(int); ok {
+				lid := uint(l)
+				labelID = &lid
+			} else {
+				return nil, nil, badRequest("label_id must be a number")
+			}
+		case "team_id":
+			if teamID, ok := v.(int); ok {
+				teamID := uint(teamID)
+				opt.TeamFilter = &teamID
+			} else {
+				return nil, nil, badRequest("team_id must be a number")
+			}
+		case "status":
+			if status, ok := v.(string); ok {
+				if fleet.HostStatus(status).IsValid() {
+					opt.StatusFilter = fleet.HostStatus(status)
+				} else {
+					return nil, nil, badRequest("status must be one of: new, online, offline, missing")
+				}
+			} else {
+				return nil, nil, badRequest("status must be a string")
+			}
+		case "query":
+			if query, ok := v.(string); ok {
```

Comment:
Trim string and check that it is not empty?
Also check that label_id above is > 0, and teamID is >= 0?

Reply:
ah good catch, i'll add support for empty strings